netstandard1.0 and uap10.0 support

Haplois opened the pull request:
Added netstandard1.0 and uap10.0 support into ObjectModel, CoreUtilities, and PlatformAbstractions

Haplois committed:
Added netstandard1.0 support

 - Added some missing enums for netstandard1.0
 - Added a Sha1 implementation for netstandard1.0
 - Added netstandard1.0 support into ObjectModel, CoreUtilities, and PlatformAbstractions

Haplois committed:
Disabled `EolTargetFramework` warning on build.

Haplois committed:
Added UAP10.0 support

nohwnd commented on scripts/build.ps1:
```diff
@@ -93,14 +93,15 @@ $env:MSBUILD_VERSION = "15.0"
-$TPB_TargetFrameworkNS2_0 = "netstandard2.0"
-$TPB_TargetFrameworkNS1_3 = "netstandard1.3"
-$TPB_TargetFrameworkNS1_0 = "netstandard1.0"
+$TPB_TargetFrameworkNS10 = "netstandard1.0"
```
Any reason to replace the less ambiguous NS2_0 naming with NS20 naming?

Haplois replied:
To keep all target framework variable names in same format.

nohwnd commented on scripts/verify-nupkgs.ps1:
```diff
@@ -19,7 +19,7 @@ function Verify-Nuget-Packages($packageDirectory)
                      "Microsoft.TestPlatform.CLI" = 350;
                      "Microsoft.TestPlatform.Extensions.TrxLogger" = 33;
-                     "Microsoft.TestPlatform.ObjectModel" = 91;
+                     "Microsoft.TestPlatform.ObjectModel" = 178;
```
What is the impact of this on the relative size of the package? Did it grow to be 2 times the size? What is the absolute size?

Haplois replied:
`16.8.0-preview-20200921-01` is 886 KB, and the unsigned package generated with this is 1142 KB; I guess we need to compare the signed versions.

Haplois committed:
Fixed build problem

Haplois committed:
Removed fill-ins for missing System.IO objects.

Haplois committed:
Removed netstandard2.0 from the packages.

Haplois committed:
Fixed some APIs in netstandard20.

Haplois committed:
Added `summary` field to nuspec files, to avoid description being truncated

Added `summary` field to nuspec files, to avoid description being truncated and avoid some build warnings.

Sanan07 commented on src/Microsoft.TestPlatform.ObjectModel/Constants.cs:
```diff
@@ -183,7 +183,12 @@ public static class Constants
         /// </summary>
-        public static readonly string DefaultResultsDirectory = Path.Combine(Directory.GetCurrentDirectory(), ResultsDirectoryName);
+        public static readonly string DefaultResultsDirectory =
+#if NETSTANDARD1_0
```
I am just curious what is the logic behind `if` statements? Why to use `.` if it is NETStandart1.0?

Haplois replied:
NETStandard 1.0 doesn't support file IO, and we can't get `Directory.GetCurrentDirectory()`; we are using `.` instead of current directory there.

Haplois committed:
Fixed assembly signing for new targets.

Haplois committed:
Fixed a typo.

Sanan07 commented on src/Microsoft.TestPlatform.ObjectModel/Framework.cs:
```diff
@@ -43,46 +47,75 @@ private Framework()
                     case "frameworkcore10":
-                        nugetFramework = CommonFrameworks.NetCoreApp10;
+                        name = CommonFrameworks.NetCoreApp10.DotNetFrameworkName;
+                        version = CommonFrameworks.Net35.Version.ToString();
```
Why `CommonFrameworks.Net35` is here? Should be `CommonFrameworks.NetCoreApp10`

Haplois replied:
Good catch! Thanks.

Haplois committed:
PR fix.

Haplois committed:
Added missing assemblies to sign.

Haplois committed:
Merged with master

ViktorHofer commented:
@Haplois I'm wondering why this was done? Was someone asking for uap or netstandard1.0 support?
